libs/features/georouting/georouting.js

```diff
@@ -0,0 +1,160 @@
+  }
+  if (pagesExist.length > 0) await Promise.all(pagesExist);
+
+  return availableLocales.filter(a => !!a);
+}
+
+function buildText(locales, config, createTag) {
+  const fragment = new DocumentFragment();
+  const wrapper = createTag('div', { class: 'text-wrapper' });
+  locales.forEach((locale) => {
+    const lang = config.locales[locale.prefix].ietf;
+    const text = createTag('p', { class: 'locale-text', lang }, locale.text);
+    wrapper.append(text);
+  });
+  fragment.append(wrapper);
+  return fragment;
+}
+
+function buildLinks(locales, config, createTag) {
+  const fragment = new DocumentFragment();
+  const wrapper = createTag('div', { class: 'link-wrapper' });
+  locales.forEach((locale) => {
+    const lang = config.locales[locale.prefix].ietf;
+    const link = createTag('a', { class: 'locale-link', lang, href: locale.url }, locale.button);
+    const para = createTag('p', { class: 'locale-link-wrapper' }, link);
+    wrapper.append(para);
+    link.addEventListener('click', () => {
+      const modPrefix = locale.prefix || 'us';
+      // set cookie so legacy code on adobecom still works properly.
+      document.cookie = `international=${modPrefix};path=/`;
+      sessionStorage.setItem("international", modPrefix);
+    });
+  });
+  fragment.append(wrapper);
+  return fragment;
+}
+
+function getCodes(data) {
+  return data.akamaiCodes.split(',').map((a) => a.toLowerCase().trim())
+}
+
+function getMatches(data, suppliedCode) {
+  return data.reduce((rdx, locale) => {
+    const localeCodes = getCodes(locale);
+    if (localeCodes.some((code) => code === suppliedCode)) rdx.push(locale);
+    return rdx;
+  }, []);
+}
+
+async function getDetails(currentPage, localeMatches, config, createTag, getMetadata) {
+  const availableLocales = await getAvailableLocales(localeMatches, config, getMetadata);
+
+  if (availableLocales && availableLocales.length > 0) {
+    currentPage.url = '#';
+    const worldIcon = createTag('img', { src: '../../../img/icons/Smock_GlobeOutline_18_N.svg', class: 'world-icon' });
+    const text = buildText([...availableLocales, currentPage], config, createTag);
+    const links = buildLinks([...availableLocales, currentPage], config, createTag);
+    const detailsFragment = new DocumentFragment();
+    detailsFragment.append(worldIcon, text, links);
+    return detailsFragment;
+  }
+  return null;
+}
+
+async function showModal(details) {
+  const { getModal } = await import('../../blocks/modal/modal.js');
+  return getModal(null, { class: 'locale-modal', id: 'locale-modal', content: details });
+}
+
+export default async function loadGeoRouting(config, createTag, getMetadata) {
+  const { locale } = config;
+
+  const urlLocale = locale.prefix.replace('/', '');
+  const storedInter = sessionStorage.getItem('international') || getCookie('international');
+  const storedLocale = storedInter === 'us' ? '' : storedInter;
+
+  const resp = await fetch(`${origin}/georouting.json`);
```

Comment:
This needs to be fixed to use the config contentRoot.

Reply:
Not to be confused with the locale.contentRoot... since there will only be one GeoRouting file.

See: https://main--stock--adobecom.hlx.page/pages/artisthub/?milolibs=georouting-v3&akamaiLocale=DE

Reply:
Yep, makes sense. 🙌